### PR TITLE
hw1 completed

### PR DESCRIPTION
**Pull request by strrust**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,138 @@
...
+                    <groupId>junit</groupId>
+                    <artifactId>junit</artifactId>
+                    <!--  pay attention, we put only groupId and artifactId, but not version  -->
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.19.0</version>
```
Comment (komyagin) [inline pom.xml:48]: You should set version property in the properties section and use it here. As you did for testng

File: pom.xml [hunk 2]
```diff
@@ -4,13 +4,138 @@
...
+            <artifactId>testng</artifactId>
+            <version>${testng.version}</version>
+            <scope>test</scope>
+            <!--  this way we can exclude transitive dependencies,
+              testng has junit in dependencies, but we don't want it in project  -->
+            <exclusions>
+                <exclusion>
+                    <groupId>junit</groupId>
+                    <artifactId>junit</artifactId>
+                    <!--  pay attention, we put only groupId and artifactId, but not version  -->
```
Comment (komyagin) [inline pom.xml:41]: I think comments are unnecessary here

File: pom.xml [hunk 3]
```diff
@@ -4,13 +4,138 @@
...
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.19.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.epam.tat.module4</groupId>
+            <artifactId>calculator</artifactId>
+            <version>1.0</version>
```
Comment (komyagin) [inline pom.xml:54]: You should set version property in the properties section and use it here. As you did for testng

File: pom.xml [hunk 4]
```diff
@@ -4,13 +4,138 @@
...
+            <systemPath>${project.basedir}/src/main/resources/calculator-1.0.jar</systemPath>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>3.1.2</version>
```
Comment (komyagin) [inline pom.xml:65]: You should set version property in the properties section and use it here. As you did for testng

File: pom.xml [hunk 5]
```diff
@@ -4,13 +4,138 @@
...
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
```
Comment (komyagin) [inline pom.xml:95]: You should set version property in the properties section and use it here. As you did for testng

File: pom.xml [hunk 6]
```diff
@@ -4,13 +4,138 @@
...
+                    <source>${java.version}</source>
+                    <target>${java.version}</target>
+                    <!--  source and target configs can be set in properties section (in same way as encoding) using
+                    ${maven.compiler.source} and ${maven.compiler.target} properties correspondingly -->
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.0.0-M5</version>
```
Comment (komyagin) [inline pom.xml:108]: You should set version property in the properties section and use it here. As you did for testng

**Review by komyagin — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 7]
```diff
@@ -4,13 +4,138 @@
...
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <testng.suite.file>${project.basedir}/src/test/resources/testngconfigs/addSubTest.xml</testng.suite.file>
+            </properties>
+        </profile>
+        <profile>
+            <id>allTests</id>
+            <properties>
+                <testng.suite.file>${testng.configs.path}/allTests.xml</testng.suite.file>
```
Comment (komyagin) [inline pom.xml:137]: This profile should be the default profile and the path is wrong. Test it

**Review by komyagin — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (strrust) [reply pom.xml]: Fixed

File: pom.xml (hunk 2 above)
Comment (strrust) [reply pom.xml]: comments removed

File: pom.xml (hunk 3 above)
Comment (strrust) [reply pom.xml]: Fixed

File: pom.xml (hunk 5 above)
Comment (strrust) [reply pom.xml]: Fixed

File: pom.xml (hunk 6 above)
Comment (strrust) [reply pom.xml]: Fixed

File: pom.xml (hunk 7 above)
Comment (strrust) [reply pom.xml]: Path changed. Profiles tested

File: pom.xml (hunk 4 above)
Comment (strrust) [reply pom.xml]: Fixed

**Review by komyagin — APPROVED** [review]

**Pull request merged**